clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -2191,16 +2383,23 @@ mlir::Value CIRGenFunction::buildAlloca(StringRef name, QualType ty,
 
 mlir::Value CIRGenFunction::buildLoadOfScalar(LValue lvalue,
                                               SourceLocation Loc) {
+  return buildLoadOfScalar(lvalue.getAddress(), lvalue.isVolatile(),
+                           lvalue.getType(), getLoc(Loc), lvalue.getBaseInfo(),
+                           lvalue.isNontemporal());
+}
+
+mlir::Value CIRGenFunction::buildLoadOfScalar(LValue lvalue,
+                                              mlir::Location Loc) {
   return buildLoadOfScalar(lvalue.getAddress(), lvalue.isVolatile(),
                            lvalue.getType(), Loc, lvalue.getBaseInfo(),
                            lvalue.isNontemporal());
 }
 
 mlir::Value CIRGenFunction::buildFromMemory(mlir::Value Value, QualType Ty) {
   // Bool has a different representation in memory than in registers.
-  if (hasBooleanRepresentation(Ty)) {
-    llvm_unreachable("NYI");
-  }
+  // if (hasBooleanRepresentation(Ty)) {
+  //   llvm_unreachable("NYI");
+  // }
```

Comment:
Ok, this is waiting #251 to get removed, let's not forget to update this once it #251 lands.